src/shared/watchdog.c

```diff
@@ -48,25 +48,42 @@ static int disable_watchdog(void) {
 }
 
 static int settimeout_watchdog(void) {
+        bool op_not_supported = false;
         usec_t t;
-        int sec;
+        int sec, r;
 
         assert(watchdog_fd > 0);
         assert(timestamp_is_set(watchdog_timeout));
 
         t = DIV_ROUND_UP(watchdog_timeout, USEC_PER_SEC);
         sec = MIN(t, (usec_t) INT_MAX); /* Saturate */
 
-        if (ioctl(watchdog_fd, WDIOC_SETTIMEOUT, &sec) < 0)
-                return log_warning_errno(errno, "Failed to set timeout to %is: %m", sec);
+        r = ioctl(watchdog_fd, WDIOC_SETTIMEOUT, &sec);
+        if (r < 0) {
```

Comment:
why introduce `r` here? you might just invoke ioctl() inside the if if check as before here...

Reply:
fixed (probably a left over)